app/src/main/java/org/fossasia/openevent/api/processor/EventListResponseProcessor.java

```diff
@@ -33,11 +37,13 @@ public void success(EventResponseList eventResponseList, Response response) {
         dbSingleton.clearDatabase(DbContract.Event.TABLE_NAME);
         dbSingleton.insertQueries(queries);
 
+        OpenEventApp.postEventOnUIThread(new EventDownloadEvent(true));
```

Comment:
What's the point of posting on the UI thread when the query insertion is being done on UI Thread itself. the `success` Callback is fired on the UI thread. You should use the CommonTaskLoop for Database updates. And then post the event on the UI thread. I thought this was explained previously.